### PR TITLE
Teature/middleware test

### PR DESCRIPTION
**Pull request by ShadowDataNerd**
Tests for the NewJWTMiddleware are still missing and need to be added.

### CODE REVIEW

**Comment by doriengr** [comment]
The tests are failing and the linter is also failing. Is it correct that this is ready for review?

File: internal/server/http/middleware/health_check_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	handler := healthcheck.New(healthcheck.Config{
+		LivenessProbe: func(_ *fiber.Ctx) bool {
+			return true
+		},
+		LivenessEndpoint: "/health",
+		ReadinessProbe: func(_ *fiber.Ctx) bool {
+			return true
+		},
+		ReadinessEndpoint: "/ready",
+	})
```
Comment (choffmann) [inline internal/server/http/middleware/health_check_test.go:26]: Do not redefine the handler here. Use the actual defined handler.

File: internal/server/http/middleware/health_check_test.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	handler := healthcheck.New(healthcheck.Config{
+		LivenessProbe: func(_ *fiber.Ctx) bool {
+			return true
+		},
+		LivenessEndpoint: "/health",
+		ReadinessProbe: func(_ *fiber.Ctx) bool {
+			return false
+		},
+		ReadinessEndpoint: "/ready",
+	})
```
Comment (choffmann) [inline internal/server/http/middleware/health_check_test.go:66]: Same here, please do not create a new handler

File: internal/server/http/middleware/requestid_test.go
Comment (choffmann) [inline internal/server/http/middleware/requestid_test.go:1]: This is the same as the logger, just using fibre middleware with no logic of its own. Although test cases are used here, many of them have nothing to do with this. I would perhaps only use the test case `should add X-Request-ID header to each request` and `should generate unique IDs for different requests`.

**Review by choffmann — CHANGES_REQUESTED** [review]
Please fix the linter and my annotations, I will work on the JWT tests

File: internal/server/http/middleware/health_check_test.go (hunk 1 above)
Comment (choffmann) [reply internal/server/http/middleware/health_check_test.go]: I just have implemented it

File: internal/server/http/middleware/health_check_test.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+	})
+
+	t.Run("should return 404 for undefined endpoint", func(t *testing.T) {
+		req := httptest.NewRequest(http.MethodGet, "/undefined", nil)
+		resp, err := app.Test(req, -1)
+
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusNotFound, resp.StatusCode)
+	})
```
Comment (choffmann) [inline internal/server/http/middleware/health_check_test.go:44]: We need tests here if the application is not ready. Please mock `AllServicesReady()` and return `false` and test it. The readiness probe should not return 200.

File: internal/server/http/middleware/health_check_test.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+func TestHealthCheck_EdgeCases(t *testing.T) {
```
Comment (choffmann) [inline internal/server/http/middleware/health_check_test.go:47]: I think we don't need that either, these are all edge cases that we don't need to test on the side of this middleware.

**Review by choffmann — CHANGES_REQUESTED** [review]
Just some more feedback on `health_check`

**Comment by choffmann** [comment]
@ShadowDataNerd I have implemented the tests for the jwt middleware. Please fix the linter and my comments :+1:

**Review by choffmann — APPROVED** [review]
